wicket-core/src/test/java/org/apache/wicket/ajax/AjaxRequestHandlerTest.java

```diff
@@ -306,6 +312,94 @@ void ajaxRedirectSetsNoCachingHeaders()
 		assertEquals("no-cache, no-store", tester.getLastResponse().getHeader("Cache-Control"));
 	}
 
+	private static class Wicket7074Application extends MockApplication
+	{
+
+		@Override
+		public Class<? extends Page> getHomePage()
+		{
+			return Wicket7074.class;
+		}
+
+
+		@Override
+		public RuntimeConfigurationType getConfigurationType() {
+			// this ise needed so that no filters are added
+			return RuntimeConfigurationType.DEPLOYMENT;
+		}
+	}
+
+	private static class WeirdException extends IllegalStateException {
+
+	}
+
+	private static class Wicket7074 extends WebPage implements IMarkupResourceStreamProvider
+	{
+		private static final long serialVersionUID = 1L;
+
+		private boolean generateError = false;
+
+		private final Component label;
+		/**
+		 * Construct.
+		 */
+		private Wicket7074()
+		{
+			setOutputMarkupId(true);
+
+			add(new AjaxLink<Void>("action")
+			{
+				private static final long serialVersionUID = 1L;
+
+				@Override
+				public void onClick(AjaxRequestTarget target)
+				{
+					// we simulate producing an exception in rendering phase
+					generateError = true;
+					target.add(label);
+				}
+			});
+
+			add(label = new Label("label", "error")
+			{
+				private static final long serialVersionUID = 1L;
+
+				@Override
+				protected void onBeforeRender()
+				{
+					super.onBeforeRender();
+					// we simulate producing an exception in rendering phase
+					if (generateError)
+					{
+						throw new WeirdException();
+					}
+				}
+			}.setOutputMarkupId(true));
+		}
+
+		@Override
+		public IResourceStream getMarkupResourceStream(MarkupContainer container,
+													   Class<?> containerClass)
+		{
+			return new StringResourceStream(
+					"<html><body><a wicket:id='action'>link1</a><br/><br/><br/><span wicket:id='label'>Link2</span></body></html>");
+		}
+	}
+
+	@Test
+	void ajaxResposeIsEmptyIfExceptionIsProducedDuringRenderingAndNoFiltersAreSet() {
+		Wicket7074Application application = new Wicket7074Application();
+		WicketTester tester = newWicketTester(application);
+		// page renders normally first time
+		tester.startPage(new Wicket7074());
+		// click on action to produce render phase error
+		assertThrows(WeirdException.class, () -> {
+			tester.clickLink("action");
+		});
+		// AJAX response is written now into a buffer => response should empty
```

Comment:
Why the buffer is not flushed ?
I'd expect that the Ajax response is still a valid XML with a redirect to the error page.